packages/astro/src/runtime/server/index.ts

```diff
@@ -282,15 +283,18 @@ function createFetchContentFn(url: URL) {
 
 // This is used to create the top-level Astro global; the one that you can use
 // Inside of getStaticPaths.
-export function createAstro(fileURLStr: string, site: string): AstroGlobalPartial {
+export function createAstro(fileURLStr: string, site: string, projectRootStr: string): AstroGlobalPartial {
   const url = new URL(fileURLStr);
+  const projectRoot = projectRootStr === '.' ? pathToFileURL(process.cwd()) : new URL(projectRootStr);
   const fetchContent = createFetchContentFn(url);
   return {
     site: new URL(site),
     fetchContent,
     // INVESTIGATE is there a use-case for multi args?
     resolve(...segments: string[]) {
-      return segments.reduce((u, segment) => new URL(segment, u), url).pathname;
+      let resolved = segments.reduce((u, segment) => new URL(segment, u), url).pathname;
+      resolved = '/' + resolved.substr(projectRoot.pathname.length);
```

Comment:
Do we want to unconditionally `substr` this? What happens if the resolved path is outside the project root?

Reply:
I'd feel more comfortable with a `replace` or `startsWith` check

Reply:
Yeah, the only time it should be outside of the project root is if someone docs `Astro.resolve('/images/penguin.png')` which would be unexpected, but seems reasonable to support so will change.

Reply:
done!